qtconsole/util.py

```diff
@@ -106,3 +106,215 @@ def get_font(family, fallback=None):
+        In [5]: all((info[k] == ideal[k] for k in ideal.keys()))
+        Out[5]: True
+    """
+    info = _find_optimal(list(map(len, items)), row_first, *args, **kwargs)
+    nrow, ncol = info["max_rows"], info["num_columns"]
+    if row_first:
+        return (
+            [
+                [
+                    _get_or_default(items, r * ncol + c, default=empty)
+                    for c in range(ncol)
+                ]
+                for r in range(nrow)
+            ],
+            info,
+        )
+    else:
+        return (
+            [
+                [
+                    _get_or_default(items, c * nrow + r, default=empty)
+                    for c in range(ncol)
+                ]
+                for r in range(nrow)
+            ],
+            info,
+        )
+
+
+def columnize(items, row_first=False, separator="  ", displaywidth=80, spread=False):
+    """Transform a list of strings into a single string with columns.
+
+    Parameters
+    ----------
+    items : sequence of strings
+        The strings to process.
+    row_first : (default False)
+        Whether to compute columns for a row-first matrix instead of
+        column-first (default).
+    separator : str, optional [default is two spaces]
+        The string that separates columns.
+    displaywidth : int, optional [default is 80]
+        Width of the display in number of characters.
+
+    Returns
+    -------
+    The formatted string.
+    """
+    if not items:
+        return "\n"
+    matrix, info = compute_item_matrix(
+        items,
+        row_first=row_first,
+        separator_size=len(separator),
+        displaywidth=displaywidth,
+    )
+    if spread:
+        separator = separator.ljust(int(info["optimal_separator_width"]))
+    fmatrix = [filter(None, x) for x in matrix]
+    sjoin = lambda x: separator.join(
+        [y.ljust(w, " ") for y, w in zip(x, info["column_widths"])]
+    )
+    return "\n".join(map(sjoin, fmatrix)) + "\n"
+
+
+def get_text_list(list_, last_sep=" and ", sep=", ", wrap_item_with=""):
+    """
+    Return a string with a natural enumeration of items
+
+    >>> get_text_list(['a', 'b', 'c', 'd'])
+    'a, b, c and d'
+    >>> get_text_list(['a', 'b', 'c'], ' or ')
+    'a, b or c'
+    >>> get_text_list(['a', 'b', 'c'], ', ')
+    'a, b, c'
+    >>> get_text_list(['a', 'b'], ' or ')
+    'a or b'
+    >>> get_text_list(['a'])
+    'a'
+    >>> get_text_list([])
+    ''
+    >>> get_text_list(['a', 'b'], wrap_item_with="`")
+    '`a` and `b`'
+    >>> get_text_list(['a', 'b', 'c', 'd'], " = ", sep=" + ")
+    'a + b + c = d'
+    """
+    if len(list_) == 0:
+        return ""
+    if wrap_item_with:
+        list_ = ["%s%s%s" % (wrap_item_with, item, wrap_item_with) for item in list_]
+    if len(list_) == 1:
+        return list_[0]
+    return "%s%s%s" % (sep.join(i for i in list_[:-1]), last_sep, list_[-1])
```

Comment:
Also I got overzealous vendoring I think this one is not necessary.

Reply:
Ok, then please remove it.